### PR TITLE
Update example and collaborator permission setting

### PR DESCRIPTION
**Pull request by salvianreynaldi**
1. To prevent people access to the created repositories from getting revoked, we need to migrate how we set permission  from `count` to `for_each`. For_each will identify elements by the given fixed string, not the dynamic index.
2. I add terraform provider block to the example
3. Upgrade to terraform 0.12 syntax

### CODE REVIEW

File: examples/simple/main.tf [hunk 1]
```diff
@@ -1,28 +1,20 @@
+provider "github" {
+  version = "2.4.1"
```
Comment (handojo1) [inline examples/simple/main.tf]: is this related to the change to for each?

Comment (salvianreynaldi) [reply examples/simple/main.tf]: Updated the PR title and description @handojo1
